### ISSUE TITLE
Important, yet missing packages from install (gnome-keyring-pam, man)

### ISSUE THREAD

**Issue by ElliotLurie** [geckolinux/geckolinux-project#142]
Hello, I really like the idea of Gecko Linux, but it is missing a few packages that would be very helpful for end users.

1) `gnome-keyring-pam`: Allows users to save passwords and such to their login keyring. Likely only needed for and might be a dependency of some GTK desktops.
2) `man`: Almost every distribution comes with the manpages preinstalled. Gecko having them would be beneficial not only for standards, but also for people needing documentation.

**Comment by geckolinux** [comment]
Hi there, thanks for the report.

1. Regarding `gnome-keyring-pam`, so for example when you add a secured WiFI network for the first time and it asks for a master password,  `gnome-keyring-pam` will make it not ask you for the master password on every login? (What I've always done is leave the password blank.) I believe this would apply only to Gnome and Cinnamon desktops.

2. Regarding `man` pages, I figure that users who actually know that man pages exist and are disposed to using them will know how to install them with YaST. The vast majority of people don't even know what they are, so I prefer to save some space.

**Comment by ElliotLurie** [comment]
Okay, if you really don't want to include the man pages that is fine. I've had trouble figuring out how to install them myself. The posix ones are easy but I can't find the ones for `zypper`, `snapper`, `bash`, etc. But that is a bit unrelated.

As for `gnome-keyring-pam`, yes it does behave in that manner. Having it would lead to a cleaner and less confusing experience for the end user.

**Comment by geckolinux** [comment]
> As for `gnome-keyring-pam`, yes it does behave in that manner. Having it would lead to a cleaner and less confusing experience for the end user.

OK, good, I'll look into this for the next release, thanks for the suggestion.

**Comment by geckolinux** [comment]
`gnome-keyring` included in relevant spins of the current set of updates.

**Comment by commonquail** [comment]
> Regarding `man` pages, I figure that users who actually know that man pages exist and are disposed to using them will know how to install them with YaST. The vast majority of people don't even know what they are, so I prefer to save some space.

I understand the motivation for this. However, I've spent at least a handful of hours recovering missing `man` pages on a GeckoLinux_NEXT_Plasma.x86_64-152.200726.0.iso installation. I had to...

1. install `man-pages`, only to find that that solved approximately none of my problems (not that I really expected that);
1. look for alternative, separate documentation packages with no luck;
1. find out where the `man-pages` project's hosted documented gets its unmanaged sources for common utilities like [`less`](https://man7.org/linux/man-pages/man1/less.1.html) and [`wget`](https://man7.org/linux/man-pages/man1/wget.1.html);
1. inspect those utilities' Leap packages to see why they seemingly didn't include the `man` pages;
1. debug those packages to figure out why those ostensibly included `man` pages did not end up installed;
1. on a lark, forcefully reinstall one of those utilities to see what would happen;
1. finally realize that all the packages with all the missing `man` pages seemingly had been correctly installed and then had their documentation removed, and that I probably had to reinstall every package to recover documentation;
1. spin up a core openSUSE Leap 15.2 VM from the DVD image so I could just copy out its `man` pages instead of literally reinstalling everything.

I don't know if there is an easy way to just (re)install all `man` pages, or to see if any installed packages are "corrupt". I do know that the only relevant Internet search results I can find all revolve around installing the `man-pages` package. I also know that if there were such a solution it would need to be extremely accessible to effectively avoid all the extra effort I went through.

Finally, I'm not certain whether this is really the expected behaviour or something went wrong for my installation. The checksums matched, though.

In case this really is working-as-intended, is there something someone in my position can do to recover with more ease, or GeckoLinux can do to support that?

**Comment by geckolinux** [comment]
Hi there, sorry for the trouble. I honestly don't know how to install the man page packages for openSUSE, as I don't need them and I don't think they're useful for most GeckoLinux. I assume that the difficulty has something to do with a package that openSUSE has marked as "recommended" that really should be marked as a required dependency for man pages to work. Since GeckoLinux doesn't installed "recommended" packages by default sometimes this exposes packaging anomalies like that.

**Comment by Unixware** [comment]
I really like your NEXT KDE but, man,  missing the man pages killing me. I can reinstall the posix and main man pages but packages that were installed before (on setup) are missing their main pages -mainly as you setup as "true" the exclude docs option on your kiwi config.

**Comment by geckolinux** [comment]
@Unixware Interesting, I actually wasn't aware why the man pages weren't automatically included. I'll look into changing that option, but I suspect it will bring a lot of bloat that I'd prefer to avoid. Most GeckoLinux users don't even know what a man page is and will just search online to find the documentation for anything.

**Comment by commonquail** [comment]
Option description; not that it helps to understand what "documentation" entails: https://osinside.github.io/kiwi/image_description/elements.html#preferences-rpm-excludedocs

Thanks, that was very useful information. This appears to also be set for the official openSUSE LEAP 15.3 (and 15.2?) WSL distribution -- I had to reinstall ~everything there, too.

I understand GeckoLinux' motivation for preferring to exclude this content, and agree that that strikes me as a reasonable default. It seems to me that the real failing lies in how KIWI...

1. ... leaves the system in an undefined state...
2. ... using a package installation configuration that is not documented by `zypper` ([`dnf` has something](https://www.mankier.com/8/dnf#--nodocs)...
3. ... without providing any information to the user...
4. ... and without an easy way to recover.

It would be instructive to learn the size difference of the distributions with and without that setting for comparison, but since there is no setting specifically for `man` pages I would not be surprised to learn that the overhead is significant.

In any case, this issue goes a long way towards recovering.

**Comment by b3taman** [comment]
**TL;DR:** man pages use very little space compared to other docs, though the latter is also not a lot.

Regarding actual disk usage, I ran the following on a pretty vanilla, recently-installed MX Linux machine (KDE version):
```
# du -shc /usr/share/doc `manpath | tr : " "`
219M	/usr/share/doc
0	/usr/local/man
4.0K	/usr/local/share/man
35M	/usr/share/man
253M	total
```
As you can see, hardly a problem for a machine which should be able to handle GeckoLinux, especially when you take this into account (run on my first GeckoLinux installation):
```
# find /usr -type d -name icon\* 2>/dev/null | xargs du -shc | tail -1
254M    total
```
So, 253 MB for man pages _plus_ other docs and 254 MB for icons.

Will more than about 250 MB be pulled in along with the man pages? Probably, but I'd still be willing to be an iconoclast ;-)

The only other time I have run into a "no man pages" situation was with a distro for embedded systems, and this really is a problem for me: no-one can remember all of `man bash`, for example.

There is another very pertinent use case: the man pages will (should) hold distro-specific information such as location of config files and options (not) compiled in. This is really important to have at hand when moving between different distros.

---

As an aside, I said elsewhere that I had recently settled upon openSUSE Leap, and the reason is that now I can use _one_ distro across Raspberry Pis, laptops, workstations, servers, and virtual servers while having a good balance between too slow and too fast updates.

Since GeckoLinux purports to be openSUSE with easier & better support for common desktop usage, I am here in the hope that the differences are not too great, hence the surprise at man page exclusion.

**Comment by b3taman** [comment]
I hasten to add: this is not criticism. I am just pointing out that the docs really aren't that big — I like pretty icons too  ;-)

**Comment by Unixware** [comment]
@b3taman you are right 100% but anyway power users use distros that suits them, others can stay eg with Chromebooks

**Comment by geckolinux** [comment]
I generally consider myself to be a "power user" and I've been using Linux almost exclusively for 20 years, but I've never found the man pages useful for my needs. I always run `command --help` for a quick overview, and if I still can't figure it out I look for examples online.
